pkgs/development/python-modules/lm-format-enforcer/default.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pydantic,
+  interegular,
+  pyyaml,
+  poetry-core,
+}:
+
+buildPythonPackage rec {
+  pname = "lm-format-enforcer";
+  version = "0.10.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "noamgat";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-GOnMj910rgzYeIeN2yLcXZDDel/Hu6nv7ov5BrlHJLg=";
+  };
+
+  build-system = [ poetry-core ];
+
+  propagatedBuildInputs = [
+    interegular
+    pydantic
+    pyyaml
+  ];
```

Comment:
`dependencies = ...`

Reply:
why is this preferred? I copied the style from outlines.

Reply:
pythonPackages now prefers `build-system` and `dependencies` instead of `nativeBuildInputs` and `propagatedBuildInputs` to match Python native packaging (e.g. pyproject.toml) style.

Reply:
Also we're hoping to stop using references (nix-support/propagated-build-inputs) for python deps eventually, to reduce # rebuilds